e catch block");
    console.log("the error is: ", err);